base/src/types.rs

```diff
@@ -106,8 +106,13 @@ pub enum Type<Id, T = AstType<Id>> {
     Record {
         /// The associated types of this record type
         types: Vec<Field<Id, Alias<Id, T>>>,
+        row: T,
+    },
+    EmptyRow,
```

Comment:
Another idea would be to have record construction be implicit, [like Elm does](https://github.com/elm-lang/elm-compiler/blob/master/src/Type/Type.hs#L46-L47).